internal/commands/pull/command_test.go

```diff
@@ -197,19 +200,109 @@ Successfully pulled app down: app
+				cmd := &Command{inputs{Project: "elsewhere", LocalPath: "app", IncludeDependencies: true}}
+
+				err := cmd.Handler(profile, ui, cli.Clients{Realm: realmClient})
+				assert.Equal(t, errors.New("something bad happened with node modules input"), err)
+
+			})
+		})
+	})
+
+	t.Run("with a realm client that successfully exports dependencies should write the archive file", func(t *testing.T) {
+		zipPkg, zipErr := zip.OpenReader("testdata/test.zip")
+		assert.Nil(t, zipErr)
+		defer zipPkg.Close()
+
+		depsPkg, err := os.Open("testdata/node_modules.zip")
+		assert.Nil(t, err)
+		defer depsPkg.Close()
+
+		var realmClient mock.RealmClient
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{{ID: "appID", Name: "appName"}}, nil
+		}
+		realmClient.FindAppFn = func(groupID, appID string) (realm.App, error) {
+			return realm.App{ID: "appID", Name: "appName"}, nil
+		}
+		realmClient.ExportFn = func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+			return "app_20210101", &zipPkg.Reader, nil
+		}
+		realmClient.ExportDependenciesArchiveFn = func(groupID, appID string) (string, io.ReadCloser, error) {
+			return "node_modules.zip", depsPkg, nil
+		}
+		realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, error) {
+			return nil, nil
+		}
+
+		message := `Saved app to disk
+Fetched dependencies as a node_modules archive
+Successfully pulled app down: app
+`
+
+		t.Run("when include node modules", func(t *testing.T) {
 			profile, teardown := mock.NewProfileFromTmpDir(t, "pull_handler_test")
 			defer teardown()
 
-			_, ui := mock.NewUI()
+			out, ui := mock.NewUI()
+			cmd := &Command{inputs{Project: "elsewhere", LocalPath: "app", IncludeNodeModules: true}}
 
+			assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: realmClient}))
+			assert.Equal(t, message, out.String())
+
+			_, pkgErr := os.Stat(filepath.Join(profile.WorkingDirectory, "app", local.NameFunctions, "node_modules.zip"))
+			assert.Nil(t, pkgErr)
+		})
+
+		t.Run("when include dependencies", func(t *testing.T) {
+			profile, teardown := mock.NewProfileFromTmpDir(t, "pull_handler_test")
+			defer teardown()
+
+			out, ui := mock.NewUI()
 			cmd := &Command{inputs{Project: "elsewhere", LocalPath: "app", IncludeDependencies: true}}
 
-			err := cmd.Handler(profile, ui, cli.Clients{Realm: realmClient})
-			assert.Equal(t, errors.New("something bad happened"), err)
+			assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: realmClient}))
+			assert.Equal(t, message, out.String())
+
+			_, pkgErr := os.Stat(filepath.Join(profile.WorkingDirectory, "app", local.NameFunctions, "node_modules.zip"))
+			assert.Nil(t, pkgErr)
 		})
+
 	})
 
-	t.Run("with a realm client that successfully exports dependencies should write the archive file", func(t *testing.T) {
+	t.Run("with a realm client that successfully exports dependencies should write the package.json", func(t *testing.T) {
```

Comment:
it could be nice to have a test that verifies the output when we `IncludePackageJSON` but actually return a `node_modules` archive

in particular, I'm interested to see what we print for the line: `"Fetched dependencies as a package.json file"` (we should probably make sure that says the right thing, i.e. "a node_modules archive")

Reply:
```suggestion
	t.Run("with a realm client that successfully exports dependencies, should write the package.json", func(t *testing.T) {
```